### PR TITLE
infra: added quarterly cache clean up (travis)

### PR DESCRIPTION
**Pull request by rnveach**
After thinking about random things late at night, and then seeing sevntu has a 2 gig cache file. Checkstyle seems to be around 600 megs.

https://app.travis-ci.com/github/sevntu-checkstyle/sevntu.checkstyle/caches
https://app.travis-ci.com/github/checkstyle/checkstyle/caches

Command from https://www.howtogeek.com/howto/ubuntu/delete-files-older-than-x-days-on-linux .

This won't be perfect as we could removing dependencies that are still used but have not updated in this timeframe.

### CODE REVIEW

**Comment by rnveach** [comment]
Run reduced the cache to 198 megs (reduced to 30% of the original).

**Comment by romani** [comment]
Interesting idea.
We definitely keep there all our old dependencies and all their depencies.

But let's make it 90 days. During 30 days we update not all depencies, but bunch of them, we can presume that during x3 time period we most likely refresh all dependencies.
Or we can take some stat of dependency commits and make it different time period.

**Comment by rnveach** [comment]
> 's make it 90 days

Done.

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by rnveach** [comment]
New push has increased the cache size by 100 megs so far.

We delete old files after the run is complete, so this means we have ~100 megs of 30 day old dependencies still in use.

**Comment by romani** [comment]
We increase size, but think it is not old dependencies, but just depencies that doesn't change much.

File: .ci/travis.sh
Comment (nrmancuso) [inline .ci/travis.sh:143]: It would be awesome if we could base cache cleaning on changed SHA of pom.xml

Comment (romani) [reply .ci/travis.sh:143]: We changing versions too often in pom.xml

Comment (rnveach) [reply .ci/travis.sh:143]: Also pom can change for other things besides version upgrades.

I will come back to this PR to unblock it.

Comment (romani) [reply .ci/travis.sh:143]: I do not recommend to block this PR, we can always improve later on.

**Review by nrmancuso — APPROVED** [review]
I would rather destroy cache completely at least once a month, but this is better than allowing accumulation of all old dependencies.

**Review by romani — APPROVED** [review]

**Pull request merged**
